### PR TITLE
Add docker registry mirror support

### PR DESCRIPTION
**Pull request by kispaljr**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
It enables the ansible playbook to set docker registry mirrors for the docker engine in the sandbox VM and also for the management kind cluster deployed by the playbook. The list of registry mirrors is expected to be in the DOCKER_REGISTRY_MIRRORS environment variable in JSON format.

We need this, because in some company intranets DockerHub access has to go through a company-wide docker registry mirror by company policy.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Enable setting docker registry mirrors in the sandbox environment.
```

### CODE REVIEW

File: e2e/provision/playbooks/cluster.yml [hunk 1]
```diff
@@ -24,6 +24,12 @@
       become: true
       when: (container_engine is not defined) or (container_engine == "docker")
       block:
+        - name: Add registry mirrors to docker configuration
+          ansible.builtin.set_fact:
+            docker_engine_daemon_json: 
+              registry-mirrors: "{{ lookup('ansible.builtin.env', 'DOCKER_REGISTRY_MIRRORS') }}"
+          when: lookup('ansible.builtin.env', 'DOCKER_REGISTRY_MIRRORS')
+
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/cluster.yml:32]: @kispaljr correct me if I'm wrong, but it seems like `registry-mirrors` and `docker_engine_daemon_json` vars are not used anywhere. Do we need to set those facts?

Comment (electrocucaracha) [reply e2e/provision/playbooks/cluster.yml:32]: Nevermind, I just noticed that is for the [`docker_engine` role](https://github.com/andrewrothstein/ansible-docker_engine/blob/v0.2.2/tasks/main.yml#L34-L50)

File: e2e/provision/playbooks/cluster.yml [hunk 2]
```diff
@@ -24,6 +24,12 @@
       become: true
       when: (container_engine is not defined) or (container_engine == "docker")
       block:
+        - name: Add registry mirrors to docker configuration
+          ansible.builtin.set_fact:
+            docker_engine_daemon_json:
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/cluster.yml:29]:
```suggestion
            docker_engine_daemon_json:
```

File: e2e/provision/playbooks/roles/bootstrap/tasks/create-mgmt.yml [hunk 3]
```diff
@@ -32,6 +32,12 @@
           extraMounts:
             - hostPath: /var/run/{{ container_engine }}.sock
               containerPath: /var/run/{{ container_engine }}.sock
+      {% if lookup('ansible.builtin.env', 'DOCKER_REGISTRY_MIRRORS')  %}
+      containerdConfigPatches:
+        - |-
+          [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+            endpoint = {{ lookup('ansible.builtin.env', 'DOCKER_REGISTRY_MIRRORS') | from_json }}
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/bootstrap/tasks/create-mgmt.yml:39]:
```suggestion
            endpoint = {{ lookup('ansible.builtin.env', 'DOCKER_REGISTRY_MIRRORS') | from_json }}
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]
@kispaljr I've suggested removing some trailing spaces to avoid issues to the Ansible linter, other than that it looks good to be merged. Don't forget to modify the installation guide to include the env var, you can take a look of this similar change https://github.com/nephio-project/docs/pull/62

File: e2e/provision/playbooks/cluster.yml (hunk 2 above)
Comment (kispaljr) [reply e2e/provision/playbooks/cluster.yml]: Fixed in da4bcb3

File: e2e/provision/playbooks/roles/bootstrap/tasks/create-mgmt.yml (hunk 3 above)
Comment (kispaljr) [reply e2e/provision/playbooks/roles/bootstrap/tasks/create-mgmt.yml]: Fixed in da4bcb3

**Comment by kispaljr** [comment]
I fixed the linter errors. and I am still trying to interpret the output of the failed tests.

**Comment by electrocucaracha** [comment]
> I fixed the linter errors. and I am still trying to interpret the output of the failed tests.

We have seen some inconsistent issues to bringing up the uesimtun0 nic which is [causing ](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/184/e2e-ubuntu-focal/1701598024968441856#1:build-log.txt%3A68650) timeouts. This issue is not related with the change you're submitting.

/lgtm

/override e2e-ubuntu-focal

**Pull request merged**